modules/gdscript/gdscript_vm.cpp

```diff
@@ -1244,7 +1244,13 @@ Variant GDScriptFunction::call(GDScriptInstance *p_instance, const Variant **p_a
 							"' to a variable of type '" + nc->get_name() + "'.";
 					OPCODE_BREAK;
 				}
-				Object *src_obj = src->operator Object *();
+
+				bool prev_free = false;
+				Object *src_obj = src->get_validated_object_with_check(prev_free);
+				if (!src_obj) {
+					err_text = vformat("Trying to assign invalid object (previously_freed = %s)", prev_free ? "true" : "false");
```

Comment:
Here we are not calling (or really doing anything) to the freed object though - it is just being assigned to another variable. This was creating an error because the debug messages were trying to access data of the previously freed object.

The problem here is that we can't validate the object type/class because it was previously freed. Should that be an error or should we just shut up like untyped variables and let the assignment happen with the freed object?

Reply:
What I meant is that this should be:
```suggestion
					if (prev_free) {
						err_text = "Trying to assign invalid previously freed instance.";
					} else {
						err_text = "Trying to assign invalid null value.";
					}
```
instead of "(previously_freed = true|false)" which is not as user friendly.

Reply:
Oh, my bad! Will fix :)